pkg/pool-weighted/test/WeightedPool.test.ts

```diff
@@ -9,13 +9,13 @@ describe('WeightedPool', function () {
   let allTokens: TokenList;
 
   const POOL_SWAP_FEE_PERCENTAGE = fp(0.01);
-  const WEIGHTS = [fp(30), fp(20), fp(15), fp(10), fp(8), fp(6), fp(4), fp(2)];
+  const WEIGHTS = Array(20).fill(fp(0.05));
```

Comment:
It is important that we use different weights, otherwise we're not testing that each token is associated with the correct weight (i.e. that there are no mistakes in the if-else chain).

We should also test different scaling facotrs (i.e. different number of decimals). Alternatively we could have 20 programatically created tests (`i = 0..19`), where we have token `i` have a different number of decimals and check that the scaling factor is correct.

Reply:
Ok; will expand on this

Reply:
It uses different weights now and, if you pass in a varyDecimals parameter, it will set each token to a different decimal. For instance, a 3-token pool would have 18,17,16; a 5-token pool would have 18,17,16,14 - and a 20 token pool represents all 18 + 0 decimal tokens.

Scaling factors are not exposed, so I had to create a MockWeightedPool to expose them and deploy that instead of the regular WeightedPool. The test now checks that the scaling factors match the predicted ones